libbeat/cmd/instance/beat.go

```diff
@@ -824,7 +824,10 @@ func (b *Beat) configure(settings Settings) error {
 		return fmt.Errorf("failed to get host information: %w", err)
 	}
 
-	fqdn, err := h.FQDN()
+	fqdnLookupCtx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
+	defer cancel()
+
+	fqdn, err := h.FQDNWithContext(fqdnLookupCtx)
```

Comment:
If this times out it falls back to the non-FQDN, does that ever get updated? Or is the Beat stuck with the non-FQDN.

If the first rule of network communication is "never make a request without a timeout", the second one is "never rely on a single network request to succeed".

Reply:
No, it doesn't get updated.  I agree that it would be nice to keep retrying but I also think that's out of scope for this PR here; this one is just about implementing the first rule.  I'll file a separate issue to implement the second one.

Reply:
https://github.com/elastic/beats/issues/37835